### ISSUE TITLE
[no-multi-assign] option to allow assign object properties?

### ISSUE THREAD

**Issue by ZYinMD** [eslint/eslint#11527]
**What rule do you want to change?**
`no-multi-assign`

**Does this change cause the rule to produce more or fewer warnings?**
fewer

**How will the change be implemented? (New option, new default behavior, etc.)?**
Add an option to allow assigning object properties on a same line

**Please provide some example code that this change will affect:**

```js
// good: 
foo1.bar = foo2.bar = foo3.bar = 'hello';

// error:
var foo = bar = 'hello'; // this is bad because it will make bar a global variable, which I believe is why this rule existed?

```
**What does the rule currently do for this code?**
warn on both cases above

**What will the rule do after it's changed?**
an option to only warn the bad case above

**Are you willing to submit a pull request to implement this change?**
I'm not capable, sorry

```js
// By the way, maybe if both foo and bar has been declared before, it's also good? 

// also good:
var foo, bar;
foo = bar = 'hello';
// This could be a separate proposal, I don't know if this will be equally easy to implement as the first one
```
